client/electron/go_vpn_tunnel.ts

```diff
@@ -333,21 +328,23 @@ class GoTun2socks {
   }
 }
 
-// Leverages the outline-go-tun2socks binary to check connectivity to the server specified in
-// `config`. Checks whether proxy server is reachable, whether the network and proxy support UDP
-// forwarding and validates the proxy credentials. Resolves with a boolean indicating whether UDP
-// forwarding is supported. Throws if the checks fail or if the process fails to start.
+/**
+ * Leverages the GoTun2socks binary to check connectivity to the server specified in `config`.
+ * Checks whether proxy server is reachable, whether the network and proxy support UDP forwarding
+ * and validates the proxy credentials.
+ *
+ * @returns A boolean indicating whether UDP forwarding is supported.
+ * @throws Error if the server is not reachable or if the process fails to start.
+ */
 async function checkConnectivity(tun2socks: GoTun2socks) {
-  try {
-    await tun2socks.checkConnectivity();
-    return true;
-  } catch (e) {
-    console.error('connectivity check error:', e);
-    if (e instanceof ProcessTerminatedExitCodeError) {
-      if (e.exitCode === ErrorCode.UDP_RELAY_NOT_ENABLED) {
-        return false;
-      }
-    }
-    throw e;
+  const output = await tun2socks.checkConnectivity();
+  // Only parse the first line, because sometimes Windows Crypto API adds warnings to stdout.
+  const outObj = JSON.parse(output.split('\n')[0]);
+  if ('tcp' in outObj && outObj.tcp) {
```

Comment:
What is the additional `x in y` check giving you? A missing key is already `undefined`, right? I think you can just do:

```suggestion
  if (outObj.tcp) {
```

Reply:
Updated